lib/MusicBrainz/Server/EditSearch/Predicate/EditNoteAuthor.pm

```diff
@@ -1,47 +1,24 @@
 package MusicBrainz::Server::EditSearch::Predicate::EditNoteAuthor;
-
 use Moose;
-use MusicBrainz::Server::Validation qw( is_database_row_id );
-
-with 'MusicBrainz::Server::EditSearch::Predicate';
-
-sub operator_cardinality_map {
-    return (
-        '=' => 1,
-        '!=' => 1,
-    );
-};
-
-sub valid {
-    my ($self) = @_;
-
-    my @args = $self->arguments;
-    return scalar(@args) == 1 && is_database_row_id($args[0]);
-}
+use namespace::autoclean;
 
-sub combine_with_query {
-    my ($self, $query) = @_;
-
-    my $sql = 'EXISTS (
+with 'MusicBrainz::Server::EditSearch::Predicate::Role::User' => {
+    template_clause => 'EXISTS (
         SELECT TRUE FROM edit_note
-         WHERE edit_note.editor = ?
+         WHERE ROLE_CLAUSE(edit_note.editor)
```

Comment:
This `ROLE_CLAUSE` stuff made my head spin a bit, but I think I understand it now. Maybe it's a bit backwards though? I wonder if we should instead be passing a subroutine here which accepts whatever arguments the role wants to give it, and constructing/returning the SQL based on those arguments. I could be missing something.